diesel/src/pg/connection/result.rs

```diff
@@ -2,21 +2,26 @@ extern crate pq_sys;
 
 use self::pq_sys::*;
 use std::ffi::CStr;
+use std::marker::PhantomData;
 use std::num::NonZeroU32;
 use std::os::raw as libc;
+use std::rc::Rc;
 use std::{slice, str};
 
 use super::raw::RawResult;
 use super::row::PgRow;
 use crate::result::{DatabaseErrorInformation, DatabaseErrorKind, Error, QueryResult};
+use crate::util::OnceCell;
 
-pub struct PgResult {
+pub(crate) struct PgResult<'a> {
     internal_result: RawResult,
     column_count: usize,
     row_count: usize,
+    column_name_map: OnceCell<Vec<Option<String>>>,
+    _marker: PhantomData<&'a super::PgConnection>,
```

Comment:
This marker seems to indicate that the connection should not be dropped or reused before the `PgResult` is dropped, but I haven't seen a reference of this in the Postgres doc. Did I miss it and the previous impl was not super safe because that was missing? Otherwise, why is this here?

Reply:
It's just there to be really safe that nothing goes wrong 🙈

Reply:
It's actually documented that it's not needed: https://www.postgresql.org/docs/13/libpq-exec.html#LIBPQ-PQCLEAR